cloud_telemetry/middleware.py

```diff
@@ -0,0 +1,71 @@
+"""Middleware for OpenTelemetry instrumentation."""
+
+from opentelemetry import metrics, trace
+from opentelemetry.exporter.otlp.proto.http.metric_exporter import OTLPMetricExporter
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.instrumentation.starlette import StarletteInstrumentor
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from starlette.applications import Starlette
+
+
+class OpenTelemetryInstrumentor:
+    """Instruments a Starlette application with OpenTelemetry."""
+
+    @staticmethod
+    def instrument_app(
+        app: Starlette,
+        service_name: str,
+        enable_tracing: bool = True,
+        enable_metrics: bool = True,
+    ):
+        """Instruments a Starlette application with OpenTelemetry.
+
+        Args:
+            app (Starlette): The Starlette application instance to be instrumented.
```

Comment:
Remove the types from pydocs unless your abolutely sure that, they won't change, better to have single source of truth in func signature.